src/manager.cpp

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#include "manager.hpp"
+
+#include <nlohmann/json.hpp>
+#include <phosphor-logging/lg2.hpp>
+
+#include <exception>
+#include <fstream>
+#include <iterator>
+#include <ranges>
+#include <string>
+#include <string_view>
+
+namespace data_sync
+{
+
+Manager::Manager(const fs::path& dataSyncCfgDir)
+{
+    parseConfiguration(dataSyncCfgDir);
+}
+
+void Manager::parseConfiguration(const fs::path& dataSyncCfgDir)
```

Comment:
I think we could add a test case for this function, but I’m okay with skipping it since the API primarily uses standard library functions without much additional logic.